datarepo-clienttests/src/main/java/scripts/testscripts/baseclasses/SimpleDataset.java

```diff
@@ -20,6 +20,7 @@
   protected TestUserSpecification datasetCreator;
   protected BillingProfileModel billingProfileModel;
   protected DatasetSummaryModel datasetSummaryModel;
+  protected boolean deleteProfile = true;
```

Comment:
This is better, but I don't understand why this isn't `private`. Renaming it may make its intent more clear
```suggestion
  protected boolean createdProfile;
```

So this is set to true after this code creates a profile, so it knows it needs to clean it up later.

Reply:
I was thinking there could potentially be a case where you wanted to create but not delete (e.g. for debugging) and that might be something you would control from the test

Reply:
YAGNI :)